torch/csrc/jit/export.cpp

```diff
@@ -110,6 +110,47 @@ void validateGraph(
   EliminateDeadCode(graph->block(), true, DCESideEffectPolicy::ALLOW_DELETING_NODES_WITH_SIDE_EFFECTS);
 }
 
+std::string GetFileRootPath(const std::string& rootPath) {
+    std::string rootPath_ = rootPath;
+    // First, making slash consistent.
+    std::replace(rootPath_.begin(), rootPath_.end(), '\\', '/');
+    // Second, remove trailing slashes, if any
+    std::regex trailer("/+$");
+    std::string root = std::regex_replace(rootPath_, trailer, std::string());
+    std::string folder = root.substr(0, root.find_last_of('/'));
+    if (folder == rootPath_) { // If no root folder specified, select cwd.
+      return std::string(".");
+    }
+    return folder;
+}
+
+std::string GetExternalFileName(const c10::optional<std::string> external_ref) {
+  auto tensorName = external_ref.value();
+  const std::string illegalChars = "\\/:?\"<>|";
+  for (int i = 0; i < tensorName.size(); i++) {
+    if (illegalChars.find(tensorName[i]) != std::string::npos) {
+        tensorName[i] = '_';
+    }
+  }
+  return tensorName;
+}
+
+void CloseFile(FILE* fp) { 
+  fclose(fp);
+}
+
+void CreateExternalFile(const at::Tensor& tensor, const std::string& tensorName,
+                        const std::string& onnx_file_path) {
+  auto folder = GetFileRootPath(onnx_file_path);
```

Comment:
Nit: do we want to make sure onnx_file_path is a file, not folder?

Reply:
So, this should work for both. In fact, we don't check explicitly for the file/folder even with existing regular model file path, because the file is not yet created. 
Anyway, I feel that this will work as expected in both cases.